From: http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/c896d26c6b7c
# HG changeset patch
# User Jan Beulich <jbeulich@novell.com>
# Date 1304931739 -3600
# Node ID c896d26c6b7c70bc82e353e7d02579fdd26af2f6
# Parent  3cb58d4cdd0b0f0ecd411804f79511cf055ec718
Subject: xen/blkfront: fix data size for xenbus_gather in connect()
Patch-mainline: n/a

From: Marek Marczykowski <marmarek@mimuw.edu.pl>

barrier variable is int, not long. This overflow caused another
variable override: "err" (in PV code) and "binfo" (in xenlinux code -
drivers/xen/blkfront/blkfront.c). The latter caused incorrect device
flags (RO/removable etc).

Signed-off-by: Marek Marczykowski <marmarek@mimuw.edu.pl>
Acked-by: Ian Campbell <ian.campbell@citrix.com>
Signed-off-by: Jan Beulich <jbeulich@novell.com>

--- sle11sp1-2011-05-20.orig/drivers/xen/blkfront/blkfront.c	2010-09-22 14:17:28.000000000 +0200
+++ sle11sp1-2011-05-20/drivers/xen/blkfront/blkfront.c	2011-05-20 13:45:03.000000000 +0200
@@ -360,7 +360,7 @@ static void connect(struct blkfront_info
 	}
 
 	err = xenbus_gather(XBT_NIL, info->xbdev->otherend,
-			    "feature-barrier", "%lu", &info->feature_barrier,
+			    "feature-barrier", "%d", &info->feature_barrier,
 			    NULL);
 	if (err)
 		info->feature_barrier = 0;
